### PR TITLE
Add gatekeeper and accept auth headers

### PR DESCRIPTION
**Pull request by jasper-jager**
An insecure endpoint is added to allow testing with the current MKS headers

### CODE REVIEW

File: src/gatekeeper.conf [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+enable-token-header: false
+enable-authorization-header: false
+enable-authorization-cookies: false
+enable-default-deny: false
+enable-refresh-tokens: true
+enable-logging: true
+preserve-host: true
+secure-cookie: true # SHOULD BE true!! Only set this to false for local dev
+server-write-timeout: 30000s # Timeout fetching complete response
+upstream-response-header-timeout: 30000s  # Time until header is received
```
Comment (JKoetsier) [inline src/gatekeeper.conf:12]: Should we remove these two timeout settings to use the default? 30000 is a bit much. (The default is 10s, that should be enough).

Comment (jasper-jager) [reply src/gatekeeper.conf:12]: Setup was copied from GOB-API, 10s should indeed be enough for GOB-StUf

File: src/gobstuf/auth/routes.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+
+def _allows_access(rule, *args, **kwargs):
+    """
+    Check access to paths with variable catalog/collection names
+    """
+    role = _get_role()
+    if role:
+        # When a role is found store the MKS USER and APPLICATION in the global object and allow acces
+        setattr(g, MKS_USER_KEY, role)
```
Comment (JKoetsier) [inline src/gobstuf/auth/routes.py:73]: Global security variables sound a bit dodgy. How global is the g variable?

Comment (jasper-jager) [reply src/gobstuf/auth/routes.py:73]: See https://flask.palletsprojects.com/en/1.1.x/appcontext/#storing-data

It's global within a context and is lost after the context ends

**Pull request merged**